### PR TITLE
IDVA5-264-client email address

### PR DESCRIPTION
**Pull request by atedros-ch**

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/controllers/personsEmailController.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { Session } from "@companieshouse/node-session-handler";
+import { ClientData } from "model/ClientData";
+import { USER_DATA } from "../utils/constants";
+import { saveDataInSession } from "../utils/sessionHelper";
+
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+    const lang = selectLang(req.query.lang);
+    const locales = getLocalesService();
+    const session: Session = req.session as any as Session;
+    const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
```
Comment (sampankumar) [inline src/controllers/personsEmailController.ts:16]: I would suggest not to do this. Just get the session data and then later suer clientData? while doing dot operation

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by ttingle-ch — APPROVED** [review]
LGTM

**Review by hemerem-ch — APPROVED** [review]

File: src/controllers/personsEmailController.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            firstName: clientData?.firstName,
+            lastName: clientData?.lastName,
+            ...pageProperties
+        });
+    } else {
+        const session: Session = req.session as any as Session;
+        const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
+
+        clientData.emailAddress = req.body["email-address"];
+        clientData.confirmEmailAddress = req.body["confirm-email"];
```
Comment (swalke-ch) [inline src/controllers/personsEmailController.ts:55]: What's the point of storing "confirm-email" in the session?

**Pull request merged**